presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -721,7 +721,7 @@ public static long levenshteinDistance(@SqlType("varchar(x)") Slice left, @SqlTy
     @SqlType(StandardTypes.BIGINT)
     public static long hammingDistance(@SqlType("varchar(x)") Slice left, @SqlType("varchar(y)") Slice right)
     {
-        int distance = 0;
+        long distance = 0;
```

Comment:
Agreed. Actually Hamming distance cannot be larger than the length of the string (same for Levenshtein, though we use the longer string here). Is it safe to assume that the length of the string cannot be larger than `INT` range? If so, then it seems that `INT` should suffice.